src/sentry/notifications/notification_action/action_handler_registry/base.py

```diff
@@ -13,6 +13,32 @@
 class IntegrationActionHandler(ActionHandler, ABC):
     provider_slug: IntegrationProviderSlug
 
+    @staticmethod
+    def get_dedup_key(action: Action) -> str:
```

Comment:
could we simplify this by making this method on the action model itself?

```
class Action(...):
    def stringify(self) -> str:
         # json string encode the config
         # json string encode the data
         # get the action.type
         # return an f string of everything together like.. f"{action.type}:{action.config}:{action.data}"
```
that way we can be really generic about it at the model layer, and keep the deduplication evaluation as a string.